beacon_node/beacon_processor/src/lib.rs

```diff
@@ -366,394 +350,104 @@ impl DuplicateCache {
-                    block,
-                    seen_timestamp,
-                    process_type,
-                    should_process,
-                },
+                work: Work::RpcBlock { process_fn },
+            },
+            ReadyWork::IgnoredRpcBlock(IgnoredRpcBlock { process_fn }) => Self {
+                drop_during_sync: false,
+                work: Work::IgnoredRpcBlock { process_fn },
             },
             ReadyWork::Unaggregate(QueuedUnaggregate {
-                peer_id,
-                message_id,
-                attestation,
-                subnet_id,
-                should_import,
-                seen_timestamp,
+                beacon_block_root: _,
+                process_fn,
             }) => Self {
                 drop_during_sync: true,
-                work: Work::UnknownBlockAttestation {
-                    message_id,
-                    peer_id,
-                    attestation,
-                    subnet_id,
-                    should_import,
-                    seen_timestamp,
-                },
+                work: Work::UnknownBlockAttestation { process_fn },
             },
             ReadyWork::Aggregate(QueuedAggregate {
-                peer_id,
-                message_id,
-                attestation,
-                seen_timestamp,
+                process_fn,
+                beacon_block_root: _,
             }) => Self {
                 drop_during_sync: true,
-                work: Work::UnknownBlockAggregate {
-                    message_id,
-                    peer_id,
-                    aggregate: attestation,
-                    seen_timestamp,
-                },
+                work: Work::UnknownBlockAggregate { process_fn },
             },
             ReadyWork::LightClientUpdate(QueuedLightClientUpdate {
-                peer_id,
-                message_id,
-                light_client_optimistic_update,
-                seen_timestamp,
-                ..
+                parent_root,
+                process_fn,
             }) => Self {
                 drop_during_sync: true,
                 work: Work::UnknownLightClientOptimisticUpdate {
-                    message_id,
-                    peer_id,
-                    light_client_optimistic_update,
-                    seen_timestamp,
+                    parent_root,
+                    process_fn,
                 },
             },
-            ReadyWork::BackfillSync(QueuedBackfillBatch { process_id, blocks }) => {
-                WorkEvent::chain_segment(process_id, blocks)
-            }
+            ReadyWork::BackfillSync(QueuedBackfillBatch(process_fn)) => Self {
+                drop_during_sync: false,
+                work: Work::ChainSegmentBackfill(process_fn),
+            },
         }
     }
 }
 
-pub struct BeaconProcessorSend<T: BeaconChainTypes>(pub mpsc::Sender<WorkEvent<T>>);
+/// Items required to verify a batch of unaggregated gossip attestations.
```

Comment:
These `...Package` structs were previously defined in the `gossip_methods.rs` file in `beacon_node/network`. I had to move them here since the `BeaconProcessor` needs to be aware of them to aggregate multiple attestations from the queue.